packages/react-file-tree/src/tree/Tree.js

```diff
@@ -101,17 +104,22 @@ export default class extends Component {
       <div style={styles.nodeContainer}
         tabIndex={'0'}
         onClick={this.handleClick.bind(this, node, nodeMetadata, index)}
+        onDoubleClick={this.handleDoubleClick.bind(this, node, nodeMetadata, index)}
+        onMouseEnter={this.handleMouseEnter.bind(this, node, nodeMetadata, index)}
+        onMouseLeave={this.handleMouseLeave.bind(this, node, nodeMetadata, index)}
         onKeyUp={this.handleKeyUp.bind(this, node, nodeMetadata, index)}
         onKeyDown={this.handleKeyDown.bind(this, node, nodeMetadata, index)}
         onContextMenu={this.handleContextMenu.bind(this, node, nodeMetadata, index)}
       >
-        <NodeComponent
-          key={path}
-          node={node}
-          metadata={nodeMetadata}
-          depth={depth}
-          index={index}
-        />
+        {
+          React.cloneElement(NodeComponent, {
```

Comment:
Lol, I've seen a few APIs take an element. I think the old Navigator used to do this? Either way, renderNode seems fine to me. Then you could render different objects depending on depth or something like that, so it seems more flexible.